### ISSUE TITLE
[Bug]: Error when running tests in a Windows folder opened with alternative casing.

### ISSUE THREAD

**Issue by BobLuursema** [microsoft/playwright#33913]
### Version

1.49.0

### Steps to reproduce

Create a new project on a Windows machine:

```
mkdir tmp
cd tmp
npm init playwright@latest
```

Run `npx playwright test` to check that the included tests work as expected.

Now change into the same folder, but with different capitalization: `cd ../TMP` and run `npx playwright test` again.

An error will appear: "Error: Playwright Test did not expect test() to be called here."

### Expected behavior

Either, I expect the tests to work since the filesystem on Windows is not case sensitive.
Alternatively, the error message could include the option that the alternative casing is the issue if the user is on Windows.

### Actual behavior

The tests do not run and the error message is pointing in the wrong direction.

### Environment

```shell
System:
    OS: Windows 10 10.0.19045
    CPU: (12) x64 Intel(R) Core(TM) i7-10850H CPU @ 2.70GHz
    Memory: 17.63 GB / 31.61 GB
  Binaries:
    Node: 18.20.5 - ~\scoop\apps\nodejs18\current\node.EXE
    npm: 10.8.2 - ~\scoop\apps\nodejs18\current\npm.CMD
    pnpm: 9.14.4 - ~\scoop\shims\pnpm.EXE
  IDEs:
    VSCode: 1.95.3 - C:\Users\M63J824\AppData\Local\Programs\Microsoft VS Code\bin\code.CMD
  Languages:
    Bash: 5.2.37 - C:\Users\M63J824\scoop\shims\bash.EXE
  npmPackages:
    @playwright/test: ^1.49.0 => 1.49.0
```

**Comment by yury-s** [comment]
These seems to be fixed in recent versions of Windows/PowerShell, I cannot reproduce it on Windows 11 neither in PowerShell nor Command Prompt, not even in Git Bash. In any case this is outside of playwright scope, since Playwright is working as intended when ran from the correct path.

**Comment by BobLuursema** [comment]
I ran into this issue first a couple months ago and it took me an hour to find out why it crashed. And I ran into it again this week but I had forgotten what the issue was so I was quite annoyed with myself for a bit 😅

I use PowerShell, I checked and I can't reproduce the issue with Command Prompt either. I wouldn't have thought that PowerShell would still be relevant here.

Anyway, thanks for looking into it! Hopefully next time I see it I will remember the problem 🙃
